cdap-ranger/cdap-ranger-lookup/src/main/java/co/cask/cdap/security/authorization/ranger/lookup/client/CDAPRangerLookupClient.java

```diff
@@ -119,6 +176,84 @@ private void initConnection() throws IOException {
     return responseData;
   }
 
+  public List<String> getResources(ResourceLookupContext context) {
+    final String userInput = context.getUserInput();
+    final String resource = context.getResourceName();
+    final Map<String, List<String>> resourceMap = context.getResources();
+    List<String> resultList = null;
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("<== CDAPResourceMgr.getResources()  UserInput: \"" + userInput + "\" resource : " + resource +
+                  " resourceMap: " + resourceMap);
+    }
+    if (userInput != null && resource != null && resourceMap != null && !resourceMap.isEmpty()) {
+      try {
+        Callable<List<String>> callableObj;
+        callableObj = new Callable<List<String>>() {
+          @Override
+          public List<String> call() {
+            List<String> retList = new ArrayList<>();
+            try {
+              List<String> list = null;
+              if (resource.trim().equalsIgnoreCase(RangerCommons.KEY_NAMESPACE)) {
+                list = getNamespaces(resourceMap.get(RangerCommons.KEY_NAMESPACE));
```

Comment:
In line 198 and line 216 we use `resourceMap.get()` value as an inclusive filter (i.e., return only those entities that have the same namespace id or application id). However, in lines 205, 208, 211, etc. we use the `resourceMap.get()` value as an exclusion.